File: kala-game-service/src/main/java/org/jesperancinha/games/kalagameservice/configuration/SecurityAdapter.java
package org.jesperancinha.games.kalagameservice.configuration;

import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.security.provisioning.JdbcUserDetailsManager;

import javax.sql.DataSource;

@Configuration
public class SecurityAdapter extends WebSecurityConfigurerAdapter {

    private final DataSource dataSource;

    private final JdbcUserDetailsManager jdbcUserDetailsManager;

    private final PasswordEncoder passwordEncoder;

    public SecurityAdapter(DataSource dataSource, JdbcUserDetailsManager jdbcUserDetailsManager, PasswordEncoder passwordEncoder) {
        this.dataSource = dataSource;
        this.jdbcUserDetailsManager = jdbcUserDetailsManager;
        this.passwordEncoder = passwordEncoder;
    }


    @Override
    protected void configure(HttpSecurity http) throws Exception {
        http.authorizeRequests()
                .antMatchers("/**")
                .authenticated()
                .and()
                .formLogin()
                .defaultSuccessUrl("http://localhost:4200")
                .and().csrf().disable();
    }

    @Override
    protected void configure(AuthenticationManagerBuilder auth) throws Exception {
        auth.userDetailsService(jdbcUserDetailsManager).passwordEncoder(passwordEncoder);
    }
}

File: kalah-game-service/src/main/kotlin/org/jesperancinha/kalah/configuration/SecurityAdapter.kt
package org.jesperancinha.kalah.configuration

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.security.provisioning.JdbcUserDetailsManager
import org.springframework.security.web.SecurityFilterChain

/**
 * Created by jofisaes on 06/09/2021
 */

@Configuration
open class SecurityAdapter(jdbcUserDetailsManager: JdbcUserDetailsManager, passwordEncoder: PasswordEncoder) {
    private val jdbcUserDetailsManager: JdbcUserDetailsManager
    private val passwordEncoder: PasswordEncoder


    @Bean
    fun filterChain(http: HttpSecurity): SecurityFilterChain = http
        .userDetailsService(jdbcUserDetailsManager)
        .authorizeRequests()
        .requestMatchers("/**")
        .authenticated()
        .and()
        .formLogin()
        .defaultSuccessUrl("http://localhost:4200")
        .and().csrf().disable().build()

    init {
        this.jdbcUserDetailsManager = jdbcUserDetailsManager
        this.passwordEncoder = passwordEncoder
    }
}
